### DISCUSSION TITLE
Sensor Refresh Error

### DISCUSSION THREAD

**Discussion by jordanphstein** [homebridge-simplisafe3/homebridge-simplisafe3#206] [Q&A] [answered]
Hello,

Ever since adding my SimpliSafe, I've noticed pretty unstable behavior while using it to automate my Apple Homekit. Here is the latest error. Does anyone have any ideas why this sensor is having this issue? I am also unsure which sensor this is...

` Sensor refresh received an error from the SimpliSafe API: TypeError: Cannot read property 'getService' of undefined
    at Object.callback (/usr/lib/node_modules/homebridge-simplisafe3/accessories/motionSensor.js:150:26)
    at /usr/lib/node_modules/homebridge-simplisafe3/simplisafe.js:911:93
    at Array.map (<anonymous>)
    at Timeout._onTimeout (/usr/lib/node_modules/homebridge-simplisafe3/simplisafe.js:911:78)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)`

**Comment by shamoon** [comment, marked as answer]
Hmm, never seen this error before. Its the motions sensor (`.../homebridge-simplisafe3/accessories/motionSensor.js`). Did you notice that you need 'secret alerts' enabled for the motion sensor to work? (plugin gives a warning: "requires secret alerts to be enabled in SimpliSafe before you can add it to Homebridge"). Im wondering if you setup the sensor and then things changed?

Either way, one thing I'd recommend to try is remove this accessory and retry. If youre still getting the error after that please post a full log output with [debug enabled](https://github.com/homebridge-simplisafe3/homebridge-simplisafe3#debug) (plugin and homebridge) and then we'll see. You can remove a single accessory from Homebridge under the settings for Config UI X ("Remove Single Cached Accessory"), if youre doing it manually its not quite as easy.

**Reply by jordanphstein** [reply]
I believe this was due to the motion notification setting on my camera!
